### ISSUE TITLE
Monero block height problem on 4 different GPU and on 3 Computers

### ISSUE THREAD

**Issue by vasssek** [xmrig/xmrig-nvidia#260]
Hi,

I've been mining new version on Monero /r. Today at block 1802223 xmrig-nvidia stopped hash calculation as you can see in the attachment. Strange thing is that it happens on 4 different GPUs (P1000, K2200, GTX1070 Ti and GTX1060 at the same block. These cards are running on 3 different computers and there is also running CPU xmrig without any problems.

I use nvidia-xmrig v. 2.14.0.

Please help me to solve this issue.

Thank you
![Nvidia P1000 Monero height 1802223 problem](https://user-images.githubusercontent.com/33999902/55280553-87484080-5327-11e9-8a59-be2c9374f37d.PNG)
![Nvidia GTX1070 + GTX 1060  Monero height 1802223 problem](https://user-images.githubusercontent.com/33999902/55280554-88796d80-5327-11e9-9fe1-e0a3a6fa22ee.PNG)
![Nvidia K2200 Monero height 1802223 problem](https://user-images.githubusercontent.com/33999902/55280555-89aa9a80-5327-11e9-9f1f-dd9b7eaa8d94.PNG)

**Comment by xmrig** [comment]
Bug confirmed.
Thank you.

**Comment by xmrig** [comment]
It NVRTC compiler bug, **only possible solution is use CUDA 10.1**, I added build with this CUDA to latest release https://github.com/xmrig/xmrig-nvidia/releases/tag/v2.14.1

Test pool to check this issue:
```json
        {
            "url": "159.65.202.177:3333",
            "user": "x+20000",
            "pass": "x",
            "rig-id": null,
            "nicehash": false,
            "keepalive": false,
            "variant": -1,
            "enabled": true,
            "tls": false,
            "tls-fingerprint": null
        },
```
It sends fake job every 20 second (time adjustable) starting from block 1802220.

**Comment by vasssek** [comment]
Thank you, wouldn't be better to add it to the next sw version, e.g. 2.14.2 ?

**Comment by xmrig** [comment]
https://github.com/xmrig/xmrig-nvidia/releases/tag/v2.14.2

@psychocrypt xmr-stak also affected by this bug.

**Comment by psychocrypt** [comment]
thanks for the notification. I will use your testpool soon and look if i
can reproduce the bug and check if we can workaround the bug

**Comment by psychocrypt** [comment]
I added a workaround for this bug in xmr-stak https://github.com/fireice-uk/xmr-stak/pull/2390

**Comment by xmrig** [comment]
https://github.com/xmrig/xmrig-nvidia/releases/tag/v2.14.3

@psychocrypt Thank you, magic `volatile` did the trick again.

**Comment by psychocrypt** [comment]
yes volatile avoid optimizations. Normally only r2 must be guarded with
volatile but to avoid similar problems in the future I disallow
optimizations for all four variables.

**Comment by vasssek** [comment]
Thank you guys 1st_place_medal:
